addons/account_payment/models/payment_provider.py

```diff
@@ -57,6 +57,7 @@ def _ensure_payment_method_line(self, allow_create=True):
             pay_method_line_same_code = self.env['account.payment.method.line'].search(
                 [
                     ('company_id', '=', self.company_id.id),
+                    # FIXME: this is likely broken, as the code is not unique and it's a related non stored field
```

Comment:
It's a related field so the field is perfectly searchable. The orm is able to make a join for it.